lib/srv/db/common/auth.go

```diff
@@ -1033,3 +1033,24 @@ func (r *elastiCacheRedisIAMTokenRequest) getSignableRequest() (*http.Request, e
 	}
 	return req, nil
 }
+
+type reportingAuth struct {
+	Auth
+	component string
+	db        types.Database
+}
+
+func GetReportingAuth(db types.Database, auth Auth) Auth {
+	return &reportingAuth{
+		Auth:      auth,
+		component: "db:auth",
+		db:        db,
+	}
+}
+
+func (r *reportingAuth) GetTLSConfig(ctx context.Context, sessionCtx *Session) (*tls.Config, error) {
+	defer methodCallMetrics("GetTLSConfig", r.component, r.db)()
+	return r.Auth.GetTLSConfig(ctx, sessionCtx)
+}
+
+var _ Auth = (*reportingAuth)(nil)
```

Comment:
This expresses the intent: `*reportingAuth` should implement `Auth`. This is trivially true now, but it may not be in the future if someone touches that code.

Reply:
Yes, this is true now. Yet a simple change may invalidate it:

```patch
type reportingAuth struct {
-	          Auth
+	auth      Auth
	component string
	db        types.Database
}
```

This is just a test; only it is written using the type system.